dhall/tests/Dhall/Test/Import.hs

```diff
@@ -70,22 +70,32 @@ successTest path = do
 
         actualExpr <- Core.throws (Parser.exprFromText mempty text)
 
-        let setCache =
-                Turtle.export "XDG_CACHE_HOME" "dhall-lang/tests/import/cache"
+        let originalCache = "dhall-lang/tests/import/cache"
+
+        let setCache = Turtle.export "XDG_CACHE_HOME"
 
         let unsetCache = Turtle.unset "XDG_CACHE_HOME"
 
         let load =
                 State.evalStateT (Test.Util.loadWith actualExpr) (Import.emptyStatus directoryString)
 
-        let runTest = do
-                if Turtle.filename (Turtle.fromText path) `elem`
-                     [ "hashFromCacheA.dhall"
-                     , "unit/asLocation/HashA.dhall"
-                     ]
-                    then do
-                        setCache
-                        _ <- load
+        let usesCache = [ "hashFromCacheA.dhall"
+                        , "unit/asLocation/HashA.dhall"
+                        , "unit/IgnorePoisonedCacheA.dhall"]
+
+        let endsIn path' = not $ null $ Turtle.match (Turtle.ends path') path
+
+        let buildNewCache = do
+                                tempdir <- Turtle.mktempdir "/tmp" "dhall-cache"
```

Comment:
It's slightly better if this use the `TMPDIR` environment variable if set (falling back to `"/tmp"` if unset), in order to interoperate better with Nix.  The reason why is that Nix sets `TMPDIR` during a build so that the build doesn't need to write outside of the temporary build directory.